### PR TITLE
HEEDLS-300 Add other section check to section query

### PR DESCRIPTION
**Pull request by benjimarshall**
### Changes
* Check whether a section has another section by amending the next section finder.
* Add completion summary information to section content.

## Testing
Amend tests to cover these new fields, and add some more tests for the other section checker. These are mostly based on the next section tests.

### CODE REVIEW

File: DigitalLearningSolutions.Data.Tests/Helpers/TutorialContentTestHelper.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public void ArchiveTutorial(int tutorialId)
             );
         }
 
-        public void UpdatePLAssessPath(int sectionId, string? newPath)
+        public void UpdatePostLearningAssessmentPath(int sectionId, string? newPath)
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Helpers/TutorialContentTestHelper.cs:26]: Thanks, nice improvement 🙂

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs [hunk 2]
```diff
@@ -993,17 +1058,352 @@ public void Get_section_content_diagnostic_status_is_true_if_one_tutorial_has_di
...
+            {
+                // Given
+                const int customisationId = 17456;
+                const int candidateId = 210934;
+                const int sectionId = 664;
+
+                const int hiddenTutorialStatus = 0;
+                const int activeDiagnosticTutorialStatus = 1;
+
+                // The hide but keep the diagnostic assessment of the tutorials in the next section, 668
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:1217]: Should this be "This" not "The"? 🙂

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs [hunk 3]
```diff
@@ -993,17 +1058,352 @@ public void Get_section_content_diagnostic_status_is_true_if_one_tutorial_has_di
...
+                // This is the only other section on this course
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:1218]: Might be easier to use a course which actually has a section with just a diagnostic assessment, such as 5694? 🙂

File: DigitalLearningSolutions.Data/Services/SectionContentService.cs [hunk 4]
```diff
@@ -27,39 +27,50 @@ public SectionContentService(IDbConnection connection, ILogger<SectionContentSer
...
+                                 OR (OtherCustomisationTutorials.DiagStatus = 1 AND OtherSections.DiagAssessPath IS NOT NULL)
+                                 OR (Customisations.IsAssessed = 1 AND OtherSections.PLAssessPath IS NOT NULL)
                             )
-                        GROUP BY
-                            CurrentSection.SectionID,
-                            NextSections.SectionID,
-                            NextSections.SectionNumber
-                        ORDER BY NextSections.SectionNumber, NextSections.SectionID
+                    ),
+                    OtherSectionsExistTable AS (
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data/Services/SectionContentService.cs:56]: Does this need to be defined as a new select? Can't the `CASE` part just be part of the select statement below? 🙂

**Review by FridaTveit — COMMENTED** [review]
A few comments, otherwise looks good 🙂

File: DigitalLearningSolutions.Data.Tests/Helpers/TutorialContentTestHelper.cs (hunk 1 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Helpers/TutorialContentTestHelper.cs:26]: Thanks!

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs (hunk 3 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs]: Yeah I agree! Done for this and the other really long tests in 2137e08 👍

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs (hunk 2 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs]: Whoops! Fixed in 2137e08 🙂

File: DigitalLearningSolutions.Data/Services/SectionContentService.cs (hunk 4 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data/Services/SectionContentService.cs:56]: Yep! Done in 2137e08 🙂

File: DigitalLearningSolutions.Data/Services/SectionContentService.cs [hunk 5]
```diff
@@ -111,7 +114,6 @@ FROM Tutorials
                             ON Applications.ApplicationID = Customisations.ApplicationID
                         INNER JOIN Sections
                             ON Sections.SectionID = Tutorials.SectionID
-                        CROSS JOIN OtherSectionsExistTable
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data/Services/SectionContentService.cs]: Should be able to remove the definition of OtherSectionsExistTable? 🙂

**Review by FridaTveit — COMMENTED** [review]
One small thing, otherwise looks good 🙂

File: DigitalLearningSolutions.Data/Services/SectionContentService.cs (hunk 5 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data/Services/SectionContentService.cs]: Yep, sorry! Removed in 3994927 🙂

**Comment by benjimarshall** [comment]
I realised I forgot a field I needed for the front end: `Applications.IncludeCertification` to decide whether to show a completion summary card, so I added this in e02f890

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**